llvm/lib/Target/AMDGPU/AMDGPUArgumentUsageInfo.h

```diff
@@ -161,6 +162,8 @@ struct AMDGPUFunctionArgInfo {
 
   // Map the index of preloaded kernel arguments to its descriptor.
   SmallDenseMap<int, KernArgPreloadDescriptor> PreloadKernArgs{};
+  // The first user SGPR allocated for kernarg preloading.
+  Register FirstKernArgPreloadReg = AMDGPU::NoRegister;
```

Comment:
```suggestion
  Register FirstKernArgPreloadReg;
```

This is already the default and we're trying to get rid of the unsigned -> Register conversion